curriculum/challenges/english/25-front-end-development/lecture-working-with-relational-databases/687ea6c021dd5913cf8fd8dd.md

```diff
@@ -0,0 +1,158 @@
+---
+id: 687ea6c021dd5913cf8fd8dd
+title: What Are Relational Databases, and How Do They Differ from Non-Relational Databases?
+challengeType: 19
+dashedName: what-are-relational-databases-and-how-do-they-differ-from-non-relational-databases
+---
+
+# --description--
+
+A relational database is a collection of data organized into tables. These tables are made of rows and columns.
+
+Rows represent individual records. For example, if you're storing user data, each row could represent a specific user.
+
+Columns represent the attributes or fields that describe each record. The table might include columns for the most important user information, including name and national ID number.
+
+One of the key characteristics of relational databases is that they allow us to connect different pieces of information by linking tables through common attributes, making it easier to see how they relate to each other.
+
+Relational databases require a schema. The schema defines the structure and organization of the database, including its tables, columns, data types, relationships, and constraints.
+
+All tables in a relational database have a primary key, which is a unique identifier for each row. Primary keys are essential for creating relationships between tables through foreign keys. You'll learn more about primary keys, foreign keys, and how they work in the coming lectures. For now, you should know that they are essential for linking records in different tables.
+
+For example, let's say that you're creating a database for an organization that runs a wildlife conservation center for critically endangered animals.
+
+One of the tables could store general information about the animals, you might call it `animals`. Each row could represent an animal, while each column could represent an attribute of that animal, like species, size, weight, age, and other unique characteristics.
+
+You should assign a unique ID to each animal to identify it in the database. To do this, you would usually add an `id` column. This would be the primary key of the table because it's unique for each animal. The ID of each animal could be used to link its record in this table to related records in other tables.
```

Comment:
```suggestion
To identify each animal uniquely, you would typically add an `id` column to the table. This would be the primary key of the table because it's unique for each animal. The ID of each animal could be used to link its record in this table to related records in other tables.
```